kernel/src/sched.rs

```diff
@@ -838,29 +838,13 @@ impl Kernel {
                     Callback::new(process.appid(), callback_id, appdata, ptr.cast())
                 });
                 let rval = platform.with_driver(driver_number, |driver| match driver {
-                    Some(Ok(d)) => {
+                    Some(d) => {
                         let res = d.subscribe(subdriver_number, callback, process.appid());
                         match res {
                             Ok(newcb) => newcb.into_subscribe_success(),
                             Err((newcb, err)) => newcb.into_subscribe_failure(err),
                         }
                     }
-                    Some(Err(d)) => {
-                        // Legacy Tock 1.x driver handling
-                        if ptr.is_some() {
-                            GenericSyscallReturnValue::Legacy(d.subscribe(
-                                subdriver_number,
-                                Some(callback),
-                                process.appid(),
-                            ))
-                        } else {
-                            GenericSyscallReturnValue::Legacy(d.subscribe(
-                                subdriver_number,
-                                None,
-                                process.appid(),
-                            ))
-                        }
-                    }
                     None => GenericSyscallReturnValue::Legacy(ReturnCode::ENOSUPPORT),
```

Comment:
It seems like we probably don't still want the `Legacy` variant here?
```suggestion
                    None => callback.into_subscribe_failure(ErrorCode::NOSUPPORT),
```

Reply:
Once this is done, I think the variant can also be removed from syscall.rs